untrusted/quote_verify/src/lib.rs

```diff
@@ -52,15 +88,53 @@ impl Quote {
         }
     }
 
-    /// Verify the quoting enclave within the report.
+    /// Verify the enclave report body within the quote.
+    pub fn verify_enclave_report_body(&self) -> Result<(), Error> {
+        let bytes = self.get_header_and_enclave_report_body();
+        let mut key = self.get_pub_key().map_err(Error::Key)?;
+        self.verify_signature(bytes, ISV_ENCLAVE_SIGNATURE_START, &mut key)
+    }
+
+    /// Verify the quoting enclave report within the quote.
     pub fn verify_quoting_enclave_report(&self) -> Result<(), Error> {
-        let signature = self.get_der_signature();
-        let report = self.get_quoting_enclave_report();
-        let hash = Sha256::digest(report);
-        let mut cert = self.get_pck_certificate()?;
-        cert.public_key_mut()
-            .verify(HashType::Sha256, &hash, &signature)
-            .map_err(Error::Signature)
+        let bytes = self.get_quoting_enclave_report();
+        let mut certificate = self.get_pck_certificate()?;
+        let key = certificate.public_key_mut();
+        self.verify_signature(bytes, QUOTING_ENCLAVE_SIGNATURE_START, key)
+    }
+
+    /// Gets the quote header and enclave report body.
+    fn get_header_and_enclave_report_body(&self) -> &[u8] {
+        &self.bytes[..QUOTE_HEADER_SIZE + ENCLAVE_REPORT_SIZE]
+    }
+
+    /// Get the public signing key for the enclave report body (and header)
+    fn get_pub_key(&self) -> Result<Pk, mbedtls::Error> {
+        let mut start = ATTESTATION_KEY_START;
+
+        // `from_binary()` is a wrapper around `mbedtls_mpi_read_binary()`, it
```

Comment:
I'm up for suggestions with regard to these comments.  They may be to verbose.
Initial development I had `unwrap()` and then when I went and converted to `?` I tried to see if I could test the logic and thus the comments.

Reply:
I don't see much value in these comments; when would a reader care about the likelihood of failure?